keanu-python/keanu/algorithm/sampling.py

```diff
@@ -142,20 +152,33 @@ def generate_samples(net: BayesNet,
     samples = samples.dropCount(drop).downSampleInterval(down_sample_interval)
     sample_iterator: JavaObject = samples.stream().iterator()
 
-    print(vertices_unwrapped)
+    all_are_scalar = _all_vertices_are_scalar(sample_from_copy)
+
     return _samples_generator(
-        sample_iterator, vertices_unwrapped, live_plot=live_plot, refresh_every=refresh_every, ax=ax)
+        sample_iterator,
+        vertices_unwrapped,
+        live_plot=live_plot,
+        refresh_every=refresh_every,
+        ax=ax,
+        all_scalar=all_are_scalar)
+
+
+def _all_vertices_are_scalar(sample_from: Iterable[Vertex]) -> bool:
+    for vertex in sample_from:
```

Comment:
Could we change this in to a call to all with a generator expression?

Reply:
what do you mean? Sorry I'm not great at python

Reply:
I'll come chat.